packages/neos-ui/src/index.js

```diff
@@ -176,6 +182,14 @@ function * application() {
             />,
         appContainer
     );
+
+    const siteNodeContextPath = yield select($get('cr.nodes.siteNode'));
+    const documentNodeContextPath = yield select($get('ui.contentCanvas.contextPath'));
+    yield put(actions.CR.Nodes.reloadState({
+        siteNodeContextPath,
+        documentNodeContextPath,
+        merge: true
+    }));
```

Comment:
I am wondering if this change actually improves performance; or if it improves *perceived* performance. As long as the nodes are not loaded, can we even interact with the UI at all?

Reply:
It improves perceived performance. You can start interacting with everything, e.g. start inline editing. The nodes are seamlessly added when loaded. It takes 100ms to add 50 nodes to the tree, so it doesn't even lag.

Reply:
It's "actual perceived performance" ;) If you read about the timings in https://github.com/neos/neos-ui/issues/1579 you get an idea. So, the perceived performance increases (you don't just see the spinner for ages), and the actual performance as well, because you can actually start doing $something.

Reply:
@kdambekalns see my RFC how to increase real performance: https://github.com/neos/neos-ui/issues/1646